### PR TITLE
make a customized bio-reader

### LINKED ISSUE #738: Support bio-data in TorchIO lazy loading
**🚀 Feature**

Currently torchio.Image (https://torchio.readthedocs.io/data/image.html#torchio.Image) can be created with paths to files that can be read by SimpleITK or nibabel, or to a directory containing DICOM files. For bio-data, different readers are needed. We could use `aicsimageio` (https://github.com/AllenCellModeling/aicsimageio).

**Motivation**

It would be great to support bio-data in TorchIO so that the large bio-image community can be engaged. 

**Pitch**

`torchio.Image` can be created with paths to files that can be read by SimpleITK or nibabel, or to a directory containing DICOM files, or any file format supported by bioformats in `aicsimageio`.

### PR DESCRIPTION
**Pull request by jxchen01**
Fixes #738 .

**Description**
support bioformat using `aicsimageio`

**Checklist**

- [ ] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [ ] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [ ] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [ ] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

File: torchio/data/image.py [hunk 1]
```diff
@@ -117,11 +118,25 @@ def __init__(
...
-        self.reader = reader
+
+        if reader is not None:
+            self.reader = reader
+        elif 'dims_bioformat' in kwargs:
+            from aicsimageio import AICSImage
+            dims = kwargs.pop('dims_bioformat')
+            bio_reader = lambda path: (
+                check_uint_to_int(
+                    AICSImage(path).get_image_dask_data(dims, **kwargs).compute()
```
Comment (JacksonMaxfield) [inline torchio/data/image.py:133]: you may want to make this use the underlying reader object.

`AICSImage(path).reader.get_image_dask_data`

`AICSImage` dumps some dimensions if they aren't part of the standard set and I don't know what the intention of this library is.

File: torchio/data/image.py [hunk 2]
```diff
@@ -117,11 +118,25 @@ def __init__(
...
-            reader: Callable = read_image,
+            reader: Callable = None,
             **kwargs: Dict[str, Any],
             ):
         self.check_nans = check_nans
-        self.reader = reader
+
+        if reader is not None:
+            self.reader = reader
+        elif 'dims_bioformat' in kwargs:
```
Comment (JacksonMaxfield) [inline torchio/data/image.py:128]: I might consider turning this block into a function itself that is static on the object.

```python
img = Image(..., reader=Image.aicsimageio_reader)
```

That way you aren't changing the defaults at all, you would just need to update the reader code to store the requested dims somewhere?

**Review by JacksonMaxfield — COMMENTED** [review]
Without much context, this is the best I have for now.

**Comment by fepegar** [comment]
I will close this and the corresponding issue (#738) for now, as a custom reader should suffice. Feel free to reopen them if necessary.

**Pull request closed without merging**